gordo_components/builder/azure_utils.py

```diff
@@ -0,0 +1,277 @@
+from typing import Any, List
+import os
+import logging
+from pytz import UTC
+from datetime import datetime
+
+from mlflow.entities import Metric, Param
+from mlflow.tracking import MlflowClient
+from azureml.core import Workspace
+from azureml.core.authentication import (
+    InteractiveLoginAuthentication,
+    ServicePrincipalAuthentication,
+)
+
+from gordo_components.builder.logging_utils import get_loggables
+
+logger = logging.getLogger(__name__)
+
+
+def _validate_dict(d: dict, name: str, required_keys: List[str]):
+    """
+    Validate the required keys are contained in provided dictionary
+
+    Parameters
+    ----------
+    d: dict
+        Dictionary to validate.
+    name: str
+        Name of dictionary to validate.
+    required_keys: List[str]
+        Keys that must be present in provided dictionary.
+    """
+    if any([key not in d for key in required_keys]):
+        raise ValueError(f"`{name}` must contain keys {', '.join(required_keys)}.")
```

Comment:
Why are we giving the dictionary a name?

Reply:
That's strictly to make the error more verbose while generalizing it to be reused.